### DISCUSSION TITLE
Question: Is there a way to cache installed updates to custom folder?

### DISCUSSION THREAD

**Discussion by azure-hu** [LegacyUpdate/LegacyUpdate#268] [Questions & Help]
I was wondering if it would be possible to store updates required by Legacy Update setup, or installed later via Legacy Update website to use it for creating slipstreamed updated installation media?

It would be okay if not every update can be backed up, e.g. .NET Framework installers, but those which can be integrated to the installer via nLite (2000, XP, or Server 2003) or vLite (Vista) could help when installation of the OS happens in an isolated environment (for example: virtualized labs with no internet connection).

I had just installed a Vista SP2 32-bit variant to a Virtual Machine, and after running Legacy Update, 594 updates were installed, which took long time. It would be much better if most of those updates could be integrated to the installer, downloading them by hand one after another takes very long time.

**Comment by kirb** [comment]
Sorry I completely missed responding to this! If you're still interested:

Files downloaded by Legacy Update setup end up in `C:\Documents and Settings\All Users\Application Data\Legacy Update` (2000/XP) or `C:\ProgramData\Legacy Update` (Vista+). They're removed as soon as the setup completes, so if you want to take that route, be ready to do a quick copy and paste of that folder. Of course it also only downloads what your system needs. You can also find the full download list at [Patches.ini](https://github.com/LegacyUpdate/LegacyUpdate/blob/main/setup/Patches.ini).

The setup also looks for these files in the same directory as itself, and will use them if they exist. That is, you can put LegacyUpdate-1.xx.exe and XPSP3.exe in the same folder, and it'll use your already-downloaded XPSP3.exe. Legacy Update 1.10 also adds a `/passive` flag. You might find this useful if you want to run Legacy Update setup at Windows install time.

As for updates downloaded via Windows Update, you can find them at `C:\Windows\SoftwareDistribution\Download`. This will prove a bit annoying, because they have random names without extensions, and most won't be a format you can just double-click and install. Rather, they'll be mostly "express" updates, which are a .cab and .psf (patch support file). The cabs contain metadata, and the psfs contain the actual update contents. There are tools that can turn these back into a regular update cab, which nLite/vLite/NTLite should be able to use for slipstreaming.

You may also consider setting up [WSUS](https://learn.microsoft.com/en-us/windows-server/administration/windows-server-update-services/get-started/windows-server-update-services-wsus) on a Windows Server, which can pull a cache of every current update, and regularly check for and download new updates. Depends which version of "annoying" you want I guess, no real straightforward answer here, other than maybe using unofficial all-in-one update packs.
